packages/core/src/modules/message-pìckup/MessagePickupModuleConfig.ts

```diff
@@ -0,0 +1,57 @@
+import type { MessagePickupProtocol } from './protocol/MessagePickupProtocol'
+import type { MessageRepository } from '../../storage/MessageRepository'
+
+/**
+ * MessagePickupModuleConfigOptions defines the interface for the options of the MessagePickupModuleConfig class.
+ * This can contain optional parameters that have default values in the config class itself.
+ */
+export interface MessagePickupModuleConfigOptions<MessagePickupProtocols extends MessagePickupProtocol[]> {
+  /**
+   * Maximum number of messages to retrieve in a single batch message pickup
+   *
+   * @default 10
+   */
+  maximumBatchSize?: number
+
+  /**
+   * Message pickup protocols to make available to the message pickup module. Only one protocol should be registered for each
+   * protocol version.
+   *
+   * When not provided, V1MessagePickupProtocol and V2MessagePickupProtocol` are registered by default.
+   *
+   * @default
+   * ```
+   * [V1MessagePickupProtocol, V2MessagePickupProtocol]
+   * ```
+   */
+  protocols: MessagePickupProtocols
+
+  /**
+   * Allows to specify a custom pickup message queue. It defaults to an in-memory repository
+   *
+   */
+  messageRepository?: MessageRepository
```

Comment:
Curious what the line is between mediator vs message pickup. Is the mediator just about routing, and is message pickup module about queuing messages? As you mentioned before, allowing messages to be queued even without mediator (so just e.g. issuer that queues messages for the holder directly) should be possible, in which case this approach makes sense.

Does the message pickup module still support `implicit` pickup? And if so, what's the protocolVersion in that case?!

Reply:
We have to dig a little bit more about it but I initially I see "Message Pickup" mainly dedicated to the _explicit_ pickup and setting up the pickup mode in general. Although the "live mode" can be also seen as an implicit mode, it can probably also handled by this module (maybe using a specific queue as suggested by @rodolfomiranda in https://github.com/hyperledger/aries-rfcs/issues/760).

The _implicit_ mode is tricky because to me it's like a particular case that works for ACA-Py but don't see it specified in an RFC. Maybe it can still be triggered from the Mediation Recipient module (as it is being done right now) or left it to be handled from outside of the framework: for instance, we can add `MediatorWebSocketConnected` and `MediatorWebSocketDisconnected` events so the consumer application can manage the reconnections as needed.